Nos/Views/AppView.swift

```diff
@@ -7,22 +7,51 @@
 
 import SwiftUI
 
+class Router: ObservableObject {
+    @Published var path = NavigationPath()
+}
+
 struct AppView: View {
     
-    @State var path = NavigationPath()
+    /// An enumeration of the navigation destinations for AppView.
+    enum Destination: String, Hashable {
+        case home
+        case relays
+        case settings
+        
+        var label: some View {
+            switch self {
+            case .home:
+                return Text(Localized.homeFeedLinkTitle.string)
+            case .relays:
+                return Text(Localized.relaysLinkTitle.string)
+            case .settings:
+                return Text(Localized.settingsLinkTitle.string)
+            }
+        }
+    }
+    
+    @EnvironmentObject var router: Router
     
     var body: some View {
-        NavigationStack(path: $path) {
+        NavigationStack(path: $router.path) {
             List {
-                NavigationLink(Localized.homeFeedLinkTitle.string) {
+                NavigationLink(value: Destination.home) { Destination.home.label }
+                NavigationLink(value: Destination.relays) { Destination.relays.label }
+                NavigationLink(value: Destination.settings) { Destination.settings.label }
+            }
+            .navigationDestination(for: Destination.self, destination: { destination in
+                switch destination {
+                case .home:
                     HomeFeedView()
-                }
-                NavigationLink(Localized.relaysLinkTitle.string) {
+                case .relays:
                     RelayView()
-                }
-                NavigationLink(Localized.settingsLinkTitle.string) {
+                case .settings:
                     SettingsView()
                 }
+            })
+            .navigationDestination(for: Event.self) { note in
```

Comment:
I'm confused why we couldn't put this code in HomeFeedView.swift, but I tried moving it there and it broke the navigation 🤷‍♂️

Reply:
I noticed that too, it's like it wants everything where you first define the NavigationStack.

Reply:
Yeah I thought in the WWDC video I watched they were putting these deeper in the view heirarchy. Something to look into later I think. It's not necessary for this PR.